packages/twenty-server/src/engine/metadata-modules/object-metadata/hooks/before-update-one-object.hook.ts

```diff
@@ -61,9 +80,48 @@ export class BeforeUpdateOneObject<T extends UpdateObjectPayload>
         };
       }
 
-      throw new BadRequestException(
-        'Only isActive field can be updated for standard objects',
-      );
+      if (
+        Object.keys(instance.update).every((key) =>
+          ['labelSingular', 'labelPlural', 'isLabelSyncedWithName'].includes(
+            key,
+          ),
+        )
+      ) {
+        if (
+          // eslint-disable-next-line no-prototype-builtins
+          instance.update.hasOwnProperty('isLabelSyncedWithName') &&
+          instance.update.isLabelSyncedWithName !== undefined
+        ) {
+          if (
+            instance.update.isLabelSyncedWithName === true &&
+            objectMetadata.standardId
+          ) {
+            const standardObjectMetadata = getStandardObjectMetadata(
+              objectMetadata.standardId,
+            );
+
+            if (standardObjectMetadata) {
+              instance.update.labelSingular =
+                standardObjectMetadata.labelSingular;
+              instance.update.labelPlural = standardObjectMetadata.labelPlural;
+            } else {
+              throw new InternalServerErrorException(
+                'Standard object not found',
+              );
+            }
+          }
+        } else {
```

Comment:
We most often try avoid using elseor elseif and favor early returns which makes the code more readable (`if x return y; if z return d;`). We also avoid nesting if structures within another